src/calico_manifests.py

```diff
@@ -375,3 +381,62 @@ def config_hash(self) -> str:
         hash = hashlib.sha256()
         hash.update(json_str.encode())
         return hash.hexdigest()
+
+    def status(self) -> FrozenSet[HashableResource]:
+        """Return all installed objects which have a `.status.conditions` attribute.
+
+        Bonus: Log events for daemonsets with unready pods.
+        """
+        installed = self.installed_resources()
+        for obj in installed:
+            if obj.kind == "DaemonSet":
+                ds: DaemonSet = obj.resource
+                if ds.status.numberReady != ds.status.desiredNumberScheduled:
+                    log_events(collect_events(self.client, obj.resource))
+        return frozenset(_ for _ in installed if _.status_conditions)
+
+    def is_ready(self, obj: HashableResource, cond: AnyCondition) -> Optional[bool]:
+        """Determine if the resource is ready."""
+        if not (ready := super().is_ready(obj, cond)):
+            log_events(collect_events(self.client, obj.resource))
+        return ready
+
+
+def by_localtime(event: Event) -> datetime.datetime:
+    """Return the last timestamp of the event in local time."""
```

Comment:
I think we need to update this docstring to specify that if the event's last timestamp is None, it returns the current time.